connector/azure-storage/src/main/java/it/pagopa/selfcare/mscore/connector/azure_storage/AzureBlobClient.java

```diff
@@ -95,6 +94,33 @@ public String getTemplateFile(String templateName) {
         }
     }
 
+    @Override
+    public File getFileAsPdf(String contractTemplate){
+        log.info("START - getFileAsPdf for template: {}", contractTemplate);
+        try{
+            final CloudBlobContainer blobContainer = blobClient.getContainerReference(azureStorageConfig.getContainer());
+            final CloudBlockBlob blob = blobContainer.getBlockBlobReference(contractTemplate);
+            BlobInputStream blobInputStream = blob.openInputStream();
+            String fileName = Paths.get(contractTemplate).getFileName().toString();
+            File downloadedFile = File.createTempFile(fileName, ".pdf");
+            FileOutputStream fileOutputStream = new FileOutputStream(downloadedFile);
+            byte[] buffer = new byte[4096];
+            int bytesRead;
+            while ((bytesRead = blobInputStream.read(buffer)) != -1) {
+                fileOutputStream.write(buffer, 0, bytesRead);
+            }
+
+            // Close the streams
+            blobInputStream.close();
+            fileOutputStream.close();
```

Comment:
you might insert .close() inside finally statement....if an exception happens the stream will not be closed